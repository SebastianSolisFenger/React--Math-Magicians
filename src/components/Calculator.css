button {
  cursor: pointer;
}

.calculator {
  display: grid;
  grid-template-columns: repeat(4, 80px);
}

.c-item {
  border-radius: 0;
  border: 1px solid gray;
  background-color: #dfdfdf;
  height: 50px;
}

.output {
  grid-column: 1 / span 4;
  background-color: #858193;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  font-size: 25px;
  color: white;
  padding-right: 10px;
}

.zero {
  grid-column: 1 / span 2;
}

.bg-orange {
  background-color: #f69510;
}
