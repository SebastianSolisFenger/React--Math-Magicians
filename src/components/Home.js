import React from 'react';

const Home = () => (
  <div className="container">
    <h1> Welcome to Math Magicians!</h1>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Numquam vitae,
      dolore minus amet laborum repellendus quia, at omnis aperiam cupiditate
      sunt. Maxime voluptatem aliquid tempore nam natus eaque pariatur expedita!
      Laboriosam tempora, illum quae dolorem amet tenetur cum ratione
      architecto, nihil iste voluptas facere error! Ex architecto quasi omnis
      harum laudantium eligendi ipsam perspiciatis dolor quae, ipsum a dicta
      corrupti!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Numquam vitae,
      dolore minus amet laborum repellendus quia, at omnis aperiam cupiditate
      sunt. Maxime voluptatem aliquid tempore nam natus eaque pariatur expedita!
      Laboriosam tempora, illum quae dolorem amet tenetur cum ratione
      architecto.
    </p>
  </div>
);

export default Home;
