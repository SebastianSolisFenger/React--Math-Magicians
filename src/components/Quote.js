import React from 'react';

const Quote = () => (
  <div className="container">
    <h2>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Atque
      repellendus temporibus minima iure quaerat deleniti deserunt quos,
      aliquam, necessitatibus illo ipsam quam perspiciatis dolor iusto. Ut
      delectus earum eaque blanditiis! Vel praesentium odit repudiandae natus
      hic debitis, modi dolores officiis molestiae obcaecati deleniti illo
      neque, cum recusandae ab inventore nostrum autem odio libero iusto quas?
      Mollitia corporis voluptates quod eveniet! Vero, corporis repellendus
      maiores, alias dolor eaque iste labore nobis quas nulla quia atque eum
      voluptate molestiae porro possimus expedita aliquid neque modi id in. Vel
      quod animi magnam inventore!
      <span className="authorname"> -- by XXXX XXXXXXXX</span>
    </h2>
  </div>
);

export default Quote;
