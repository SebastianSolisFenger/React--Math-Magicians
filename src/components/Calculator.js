import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [obj, updateObj] = useState({ total: 0, next: null, operation: null });
  const clickBtn = (e) => {
    updateObj(calculate(obj, e.target.id));
  };

  const { total, next, operation } = obj;
  return (
    <div className="calculator-Container">
      <h2 className="defaultOutput box">
        <span>{total}</span>
        <span>{operation}</span>
        <span>{next}</span>
      </h2>
      <button type="button" onClick={clickBtn} id="AC" className="box bt">
        AC
      </button>
      <button type="button" onClick={clickBtn} id="+/-" className="box bt">
        +/-
      </button>
      <button type="button" onClick={clickBtn} id="%" className="box bt">
        %
      </button>
      <button
        type="button"
        onClick={clickBtn}
        id="รท"
        className="box bg-orange bt"
      >
        รท
      </button>
      <button type="button" onClick={clickBtn} id="7" className="box bt">
        7
      </button>
      <button type="button" onClick={clickBtn} id="8" className="box bt">
        8
      </button>
      <button type="button" onClick={clickBtn} id="9" className="box bt">
        9
      </button>
      <button
        type="button"
        onClick={clickBtn}
        id="x"
        className="box bg-orange bt"
      >
        x
      </button>
      <button type="button" onClick={clickBtn} id="4" className="box bt">
        4
      </button>
      <button type="button" onClick={clickBtn} id="5" className="box bt">
        5
      </button>
      <button type="button" onClick={clickBtn} id="6" className="box bt">
        6
      </button>
      <button
        type="button"
        onClick={clickBtn}
        id="-"
        className="box bg-orange bt"
      >
        -
      </button>
      <button type="button" onClick={clickBtn} id="1" className="box bt">
        1
      </button>
      <button type="button" onClick={clickBtn} id="2" className="box bt">
        2
      </button>
      <button type="button" onClick={clickBtn} id="3" className="box bt">
        3
      </button>
      <button
        type="button"
        onClick={clickBtn}
        id="+"
        className="box bg-orange bt"
      >
        +
      </button>
      <button type="button" onClick={clickBtn} id="0" className="box zero bt">
        0
      </button>
      <button type="button" onClick={clickBtn} id="." className="box bt">
        .
      </button>
      <button
        type="button"
        onClick={clickBtn}
        id="="
        className="box bg-orange bt"
      >
        =
      </button>
    </div>
  );
};

export default Calculator;
