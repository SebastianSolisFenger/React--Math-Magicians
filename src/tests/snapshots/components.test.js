import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../../components/Home';
import Quote from '../../components/Quote';
import Calculator from '../../components/Calculator';

describe('The Display of the webpage', () => {
  it('Display Home page', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Display Quote page', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Display Calculator page', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
