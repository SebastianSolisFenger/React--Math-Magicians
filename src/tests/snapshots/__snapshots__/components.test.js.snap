// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Display of the webpage Display Calculator page 1`] = `
<div
  className="calculator-Container "
>
  <h1
    className="letsDo"
  >
    Let's Do Some Math
  </h1>
  <div
    className="app"
  >
    <h2
      className="defaultOutput box"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </h2>
    <button
      className="box btn"
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="box btn"
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="box btn"
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="box bg-orange btn"
      id="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="box btn"
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="box btn"
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="box btn"
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="box bg-orange bt"
      id="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="box btn"
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="box btn"
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="box btn"
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="box bg-orange btn"
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="box btn"
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="box btn"
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="box btn"
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="box bg-orange btn"
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="box zero btn"
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="box btn"
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="box bg-orange btn"
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`The Display of the webpage Display Home page 1`] = `
<div
  className="container"
>
  <h1>
     Welcome to Math Magicians!
  </h1>
  <p>
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Numquam vitae, dolore minus amet laborum repellendus quia, at omnis aperiam cupiditate sunt. Maxime voluptatem aliquid tempore nam natus eaque pariatur expedita! Laboriosam tempora, illum quae dolorem amet tenetur cum ratione architecto, nihil iste voluptas facere error! Ex architecto quasi omnis harum laudantium eligendi ipsam perspiciatis dolor quae, ipsum a dicta corrupti!
  </p>
  <p>
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Numquam vitae, dolore minus amet laborum repellendus quia, at omnis aperiam cupiditate sunt. Maxime voluptatem aliquid tempore nam natus eaque pariatur expedita! Laboriosam tempora, illum quae dolorem amet tenetur cum ratione architecto.
  </p>
</div>
`;

exports[`The Display of the webpage Display Quote page 1`] = `
<div
  className="container"
>
  <h2
    className="Quote-P"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  </h2>
  <span
    className="authorname"
  >
     -- by William Paul Thuston
  </span>
</div>
`;
